### PR TITLE
Fix tests of continuous deployment

### PR DESCRIPTION
**Pull request by fpoli**

### CODE REVIEW

File: prusti-tests/tests/compiletest.rs [hunk 1]
```diff
@@ -11,17 +11,31 @@ use prusti_server::ServerSideService;
...
+            ["..", "target", "debug", "prusti-rustc"].iter().collect()
+        }
     } else {
-        ["..", "target", "debug", "prusti-rustc"].iter().collect()
+        if cfg!(windows) {
+            ["..", "target", "release", "prusti-rustc.exe"].iter().collect()
+        } else {
+            ["..", "target", "release", "prusti-rustc"].iter().collect()
+        }
     };
```
Comment (Aurel300) [inline prusti-tests/tests/compiletest.rs]:
```rust
let target_directory = ...;
let executable_name = ...;
let local_prusti_rustc_path: PathBuf = ["target", target_directory, executable_name].iter().collect();
let workspace_prusti_rustc_path: PathBuf = ["..", "target", target_directory, executable_name].iter().collect();
```

Comment (fpoli) [reply prusti-tests/tests/compiletest.rs]: Feel free to push the change; I'm going to lunch :smile:

**Comment by Aurel300** [comment]
Locally this crashed for me when testing `borrow_in_loop.rs`?

**Comment by fpoli** [comment]
> Locally this crashed for me when testing `borrow_in_loop.rs`?

That test works fine for me. In case you determine that it's a flaky test I suggest to disable (or fix) it, to make CI tests reliable.

Btw, I merged `test.yml` with `deploy.yml`, to be sure that we deploy only when all tests are passing. The file is getting a bit long but it seems the best way.

**Comment by Aurel300** [comment]
Are you testing in `release` mode? When I try to run Prusti on the file manually, it just exits with exit code 255, without any message. (Just to be clear, it crashes for me in `debug` mode, currently compiling a `release` build.)

**Comment by fpoli** [comment]
> Are you testing in `release` mode? When I try to run Prusti on the file manually, it just exits with exit code 255, without any message.

The following works for me:
```
./x.py build --release
./x.py test --release borrow_in_loop # Ok
./target/release/prusti-rustc --edition=2018 -Awarnings prusti-tests/tests/filter/pass/borrow_in_loop.rs # Ok
```

**Comment by Aurel300** [comment]
Hm no, it crashes with exit code 255 for me both on `debug` and `release` builds.

**Comment by fpoli** [comment]
> Hm no, it crashes with exit code 255 for me both on `debug` and `release` builds.

Strange... Since when? What's the log, if there is any? I suggest to open an issue if you find something to discuss.

**Comment by Aurel300** [comment]
This is probably since the dependencies update. As I said there is no log (even with backtrace enabled). It just says "Verification of 3 items", then exits. I'll investigate to see where it actually crashes.

**Comment by fpoli** [comment]
Did you enable the log with the environment variable `PRUSTI_LOG=trace`? If there is no stacktrace it might be that the JVM is crashing, so you could try to enable core dumps.

**Comment by Aurel300** [comment]
```
$ PRUSTI_LOG=trace RUST_BACKTRACE=1 ./target/release/prusti-rustc --edition=2018 -Awarnings prusti-tests/tests/filter/pass/borrow_in_loop.rs
  __          __        __  ___             
 |__)  _\/_  |__) |  | /__`  |   ____\/_  | 
 |      /\   |  \ \__/ .__/  |       /\   | 

{commit 061a2a7 2020-10-01 12:40:44 UTC, built on 2020-10-01 13:05:03 UTC}


Verification of 3 items...
[2020-10-01T18:36:15Z INFO  prusti_viper::verifier] Received 3 functions to be verified:
[2020-10-01T18:36:15Z INFO  prusti_common::stopwatch::log_level] [prusti-viper] Starting: encoding to Viper
[2020-10-01T18:36:15Z INFO  prusti_viper::verifier]  - take from prusti-tests/tests/filter/pass/borrow_in_loop.rs:1:1: 1:27 (#0) (borrow_in_loop::take)
[2020-10-01T18:36:15Z INFO  prusti_viper::verifier]  - test from prusti-tests/tests/filter/pass/borrow_in_loop.rs:3:1: 3:10 (#0) (borrow_in_loop::test)
[2020-10-01T18:36:15Z INFO  prusti_viper::verifier]  - main from prusti-tests/tests/filter/pass/borrow_in_loop.rs:11:1: 11:10 (#0) (borrow_in_loop::main)
[2020-10-01T18:36:15Z INFO  prusti_viper::encoder::encoder] Encoding: take from prusti-tests/tests/filter/pass/borrow_in_loop.rs:1:1: 1:27 (#0) (borrow_in_loop::take)
```

That is all it produced in `release` mode. In `debug` this follows:

```
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::encoder] Collecting closure instantiations...
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] Collecting closure instantiations in mir DefId(0:4 ~ borrow_in_loop[317d]::test)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] Collecting closure instantiations in mir DefId(0:5 ~ borrow_in_loop[317d]::main)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] Collecting closure instantiations in mir DefId(0:3 ~ borrow_in_loop[317d]::take)
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::encoder] closure_instantiations: {}
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] encode_builtin_method_def(HavocBool)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] encode_builtin_method_def(HavocInt)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] encode_builtin_method_def(HavocRef)
[2020-10-01T18:37:33Z INFO  prusti_viper::encoder::encoder] Encoding: take from prusti-tests/tests/filter/pass/borrow_in_loop.rs:1:1: 1:27 (#0) (borrow_in_loop::take)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] is_trusted DefId(0:3 ~ borrow_in_loop[317d]::take)
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::encoder] is_trusted DefId(0:3 ~ borrow_in_loop[317d]::take) = false
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::encoder] encode_procedure(DefId(0:3 ~ borrow_in_loop[317d]::take))
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::procedure] Encoding procedure DefId(0:3 ~ borrow_in_loop[317d]::take)
[2020-10-01T18:37:33Z DEBUG prusti_interface::environment::procedure] spec basic blocks: {}
[2020-10-01T18:37:33Z DEBUG prusti_interface::environment::loops] ordered_blocks: [bb0]
[2020-10-01T18:37:33Z DEBUG prusti_interface::environment::loops] loop_exit_blocks: {}
[2020-10-01T18:37:33Z DEBUG prusti_interface::environment::loops] nonconditional_loop_blocks: {}
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::procedure_encoder] ProcedureEncoder constructor
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] MirEncoder constructor
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] run join_operation=Union
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] merge_effects bb=bb0
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] apply_statement_effects location=bb0[0]
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] apply_terminator_effects bb=bb0
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] apply_terminator_effects bb=bb0
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] run join_operation=Intersect
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] merge_effects bb=bb0
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] apply_statement_effects location=bb0[0]
[2020-10-01T18:37:33Z TRACE prusti_interface::environment::mir_analyses::initialization] [enter] apply_terminator_effects bb=bb0
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] Encode place _1
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate name '&mut std::boxed::Box<i32>'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate name 'std::boxed::Box<i32>'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate name 'i32'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate 'i32'
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::type_encoder] Encode value field for type 'i32'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate 'std::boxed::Box<i32>'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate '&mut std::boxed::Box<i32>'
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] Encode place _1
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] Encode place _0
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate name '()'
[2020-10-01T18:37:33Z DEBUG prusti_viper::encoder::type_encoder] Encode type predicate '()'
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] Encode place _0
[2020-10-01T18:37:33Z TRACE prusti_viper::encoder::mir_encoder] Encode place _1
```

**Comment by fpoli** [comment]
Good, so you see at least INFO messages. TRACE and DEBUG are disabled in release mode, I forgot about it:

https://github.com/viperproject/prusti-dev/blob/d0fbc96b95461c20fe2c174910ac9ed722f59ef0/prusti-viper/Cargo.toml#L14

**Comment by Aurel300** [comment]
Ok, sorry, false alarm. It was actually exiting with `255` on any test that required the verifier. It seems I had to recompile and then use `./x.py`, as running the `rustc` binary directly doesn't set up the correct environment variables. Unsure why it wouldn't even report the error/backtrace; I guess it's a panic from a different thread?

**Comment by Aurel300** [comment]
Ok, I hit this issue again. This is getting a bit bizarre. If I put an `info!("...");` (with any message) in `Verifier::verify`, it works. If not, it exists with exitcode 255. 😱

**Comment by fpoli** [comment]
Some ideas for debugging:
* Try `eprintln!(..)` instead of `info!(..)`.
* Try running Prusti in a debugger, to find out from where the `exit(255)` comes from.
* Try running `prusti-driver` directly, with the same arguments and environment variables that `prusti-rustc` uses.
* Try using the Prusti server, by setting the `PRUSTI_SERVER_ADDRESS` variable.
* Try deleting one file by one from `target`, until recompiling+running Prusti works.
* Check what were the [latest system calls](https://stackoverflow.com/questions/31045575/how-to-trace-system-calls-of-a-program-in-mac-os-x), to see whether any of them failed in some strange way.

I also suggest to take a copy of `target/` before trying stuff, for reproducibility.

**Pull request merged**